EEJT-0002/src/main/java/com/epam/esm/giftcertficate/repository/GiftCertificateRepositoryImp.java

```diff
@@ -0,0 +1,140 @@
+package com.epam.esm.giftcertficate.repository;
+
+import com.epam.esm.giftcertficate.model.GiftCertificate;
+import com.epam.esm.tag.model.Tag;
+import com.epam.esm.utils.query.SqlQuery;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.time.Instant;
+import java.util.*;
+
+@Repository
+public class GiftCertificateRepositoryImp implements GiftCertificateRepository {
+
+    private final JdbcTemplate jdbcTemplate;
+    private static final String ID = "id";
+    private static final String NAME = "name";
+    private static final String DESCRIPTION = "description";
+    private static final String PRICE = "price";
+    private static final String DURATION = "duration";
+    private static final String CREATE_DATE = "create_date";
+    private static final String LAST_UPDATE_DATE = "last_update_date";
+    private static final String PART_OF_QUERY_FOR_UPDATING = "UPDATE gift_certificate SET ";
+    private static final int DB_OPERATION_SUCCESSFUL_RESULT = 1;
+
+    @Autowired
+    public GiftCertificateRepositoryImp(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private List<Tag> getAllTagsByCertificateId(long certificateId) {
+        return jdbcTemplate.query(SqlQuery.GiftCertificate.GET_ALL_TAGS_BY_CERTIFICATE_ID, new Long[]{certificateId},
+                (resultSet, i) ->
+                        new Tag().setId(resultSet.getLong(ID)).setName(resultSet.getString(NAME)));
+    }
+
+    @Override
+    public List<GiftCertificate> getAllGiftCertificates() {
+        return jdbcTemplate.query(SqlQuery.GiftCertificate.GET_ALL_CERTIFICATES, (resultSet, i) ->
+                new GiftCertificate().setId(resultSet.getLong(ID)).
+                        setName(resultSet.getString(NAME)).
+                        setTags(getAllTagsByCertificateId(resultSet.getLong(ID))).
+                        setDescription(resultSet.getString(DESCRIPTION)).
+                        setPrice(resultSet.getInt(PRICE)).
+                        setDuration(resultSet.getInt(DURATION)).
+                        setCreateDate(resultSet.getDate(CREATE_DATE)).
+                        setLastUpdateDate(resultSet.getDate(LAST_UPDATE_DATE)));
+    }
+
+    @Override
+    public boolean createCertificate(GiftCertificate giftCertificate) {
+        Instant currentTime = Instant.now();
+        return jdbcTemplate.update(SqlQuery.GiftCertificate.CREATE_CERTIFICATE,
+                giftCertificate.getName(),
+                giftCertificate.getDescription(),
+                giftCertificate.getPrice(),
+                giftCertificate.getDuration(),
+                currentTime,
+                currentTime) == DB_OPERATION_SUCCESSFUL_RESULT;
```

Comment:
You are getting closer. But passing currentTime twice still makes not a lot of sense.
What I wanterd is to rename first one to creationDate, and second one to updatedDate or smth similar. Which will expain with the naming the purpose of two `Instant.now()` usages

Reply:
done